Adding GVRObjectPicker, fixed bug with hit position

NolaDonato opened the pull request:
This is the same as PR #826 with commits squashed
This patch fixes a bug with the calculation of the hit location for
 GVRMeshCollider
 It add GVRObjectPicker which picks against the bounding box of the scene
 object the picker is attached to.

Updated documentation on picking.

GearVRF DCO signed off by: Nola Donato nola.donato@samsung.com

rahul27 commented:
I still haven't found the time to try this out. I should get to it next week. 

Am okay if someone else wants to try this out and :+1: it.

rahul27 commented:
@NolaDonato did you use the test framework to test this code or did you modify an existing sample?

I was planning on writing some code to try it, would help if you already have something.

NolaDonato commented:
I modified a sample but I did not test really thoroughly. I will write some unit tests when I am done fixing shadow mapping.

________________________________
From: rahul <notifications@github.com>
Sent: Monday, August 29, 2016 4:27 PM
To: Samsung/GearVRf
Cc: Nola Donato; Mention
Subject: Re: [Samsung/GearVRf] Adding GVRObjectPicker, fixed bug with hit position (#829)


@NolaDonato<https://github.com/NolaDonato> did you use the test framework to test this code or did you modify an existing sample?

I was planning on writing some code to try it, would help if you already have something.

-
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/Samsung/GearVRf/pull/829#issuecomment-243288347>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AQw1S_mP-sOor0jIEDL2QqzVNnOhe9jTks5qk2rIgaJpZM4JrgF5>.

liaxim commented on GVRf/Framework/framework/src/main/java/org/gearvrf/GVRObjectPicker.java:
```diff
@@ -0,0 +1,134 @@
+     * @see IPickEvents
+     * @see GVRFrustumPicker.pickVisible
+     */
+    public void doPick()
```
Regarding the javadoc: if not attached to a scene object, wouldn't it still pick against the list of visible colliders? How the pick ray comes into play in this case?

NolaDonato replied:
If not attached to a scene object, all of the pickers act like they are attached to the owner of the camera rig. The pick ray is always relative to that. Maybe we should just get rid of the pick ray. The reason I did not do that is in case the IO devices needed something different than that.

liaxim commented:
Looks good to me

liaxim commented:
Any comments? Will merge tomorrow otherwise.

rahul27 commented:
I modified the eye picking sample to test the object picker. 

Picking works fine with the bunnies, but does not seem to work well with the quads in the scene. Here is the modified sample I used to test:

https://github.com/rahul27/GearVRf-Demos/commit/e5cc9442b9bd37629e0e0e740b93ef86f563b170

Use the swipe to move the cursor forward/backward.

liaxim commented:
@NolaDonato Can this be reviewed and tested once more?

liaxim commented:
@rahul27 I see a number of commits since your comment. Would you mind running your test case again? Thanks.

rahul27 commented:
Yup. Still see the issue.

NolaDonato commented:
I fixed the problem Rahul found. I think there might be a problem with our bounding box computation logic but using GVRSceneObject.isColliding seems to get around it.

liaxim commented:
This now has 55 commits. A lot of them that do not belong. Impossible to review. Please shrink the number of commits down the actual changes. Thank you.

NolaDonato commented:
Yeah I guess I made a mess of it when I committed from my personal laptop. I forgot my power adapter last night and I wanted to fix it so I did it on a different machine. I thought I pulled over everything before I started so I'm not sure what happened. I used the method you suggested to force push just mychanges  (git push origin HEAD:objectpicker2 --force)

rahul27 commented:
@NolaDonato I've reverted the picker to use the bounding volume and use the modified matrix transformation here https://github.com/Samsung/GearVRf/pull/862.

Can we close this pull request?

NolaDonato commented:
yes
